frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RenameBackgroundable.java

```diff
@@ -0,0 +1,117 @@
+package com.virtuslab.gitmachete.frontend.actions.backgroundables;
+
+import static com.virtuslab.gitmachete.frontend.common.WriteActionUtils.runWriteActionOnUIThread;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.nio.file.Path;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.GitLocalBranch;
+import git4idea.repo.GitRepository;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.gitmachete.frontend.file.MacheteFileWriter;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+public class RenameBackgroundable extends Task.Backgroundable {
+
+  protected final Project project;
+  protected final GitRepository gitRepository;
+  protected final BranchLayout branchLayout;
+  protected final IBranchLayoutWriter branchLayoutWriter;
+  protected final Runnable renameRunnable;
+  protected final String currentBranchName;
+  protected final String newBranchName;
+
+  public RenameBackgroundable(
```

Comment:
Probably this entire backgroundable class is redundant. `git4idea.branch.GitBrancher#renameBranch` itself runs in background, and hence it can probably be executed directly on UI thread. No need to wrap `#renameBranch` call in yet another `RenameBackgroundable`